File: 0056-merge-intervals/0056-merge-intervals.cpp
#include <bits/stdc++.h>
#include <iostream>
#include <iomanip>
#include <array>
#include <vector>
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
    int n=intervals.size()-1;
    sort(intervals.begin(), intervals.end());

    vector<vector<int>> v2;
    vector<int> temp = intervals[0];

    for(int i=0; i<=n; i++) {
        if(temp[1] >= intervals[i][0]) {
            temp[1] = max(temp[1], intervals[i][1]);
        }
        else {
            v2.push_back(temp);
            temp=intervals[i];
        }
    }
    v2.push_back(temp);
    return v2;
    }
};

File: 0056-merge-intervals/0056-merge-intervals.java
class Solution {
    public int[][] merge(int[][] intervals) {

        Arrays.sort(intervals, (a, b) -> Integer.compare(a[0], b[0]));
        
        int result[][] = new int[intervals.length][2];
        int[] current = intervals[0];
        result[0] = current;
        int index = 0;

        for(int[] interval: intervals) {
            int currentEnd = current[1];
            int nextBegin = interval[0];
            int nextEnd = interval[1];

            if(currentEnd >= nextBegin) {
                current[1] = Math.max(currentEnd, nextEnd);
            }
            else {
                current = interval;
                result[++index] = current;
            }
        }
        return Arrays.copyOf(result, index + 1);
    }
}